### ISSUE TITLE
vsphere_virtual_machine.vm: error reconfiguring virtual machine: error processing network device changes post-clone: network_interface.0: ServerFaultCode: The object or item referred to could not be found.

### ISSUE THREAD

**Issue by den-is** [terraform-providers/terraform-provider-vsphere#521]
I'm trying to create VM out of vm-template with DHCP "configuration".
By default vm-template has DHCP enabled.
Same vm-template is used for creating VMs in cluster where all machines have static IP, configured using `network_interface`. Everything works just fine.
Now when I'm leaving `network_interface {}` stanza, in clone config, empty, so DHCP , terraform throws error seen in debug output below.
Another difference is that I'm deploying this DHCP enabled VMs in different cluster.
But both static-cluster and dhcp-cluster are members of the same vCenter/DC.

I just can't see places where to look at, to identify problem, if I'm missing something.

I can see that vmware successfully deploys/clones VM and in the end when error occurs, VMware just deletes newly created VM.

### Terraform Version
```
Terraform v0.11.7
+ provider.vsphere v1.4.1
vCenter, esxi v6.5
```

### Terraform Configuration Files
```hcl
data "vsphere_datacenter" "dc" {
  name = "${var.datacenter}"
}

data "vsphere_resource_pool" "rpool" {
  name          = "${var.cluster}/Resources"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}

data "vsphere_datastore" "ds" {
  name          = "${var.datastore}"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}

data "vsphere_network" "net" {
  name          = "${var.network}"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}

data "vsphere_virtual_machine" "template" {
  name          = "${var.template}"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}

resource "vsphere_virtual_machine" "vm" {
  count            = "${var.servers}"
  name             = "${var.hostname_prefix}-${format("%02d", count.index + 1)}.${var.domain}"
  resource_pool_id = "${data.vsphere_resource_pool.rpool.id}"
  datastore_id     = "${data.vsphere_datastore.ds.id}"
  num_cpus         = "${var.cpu}"
  memory           = "${var.memory}"
  guest_id         = "${data.vsphere_virtual_machine.template.guest_id}"
  scsi_type        = "${data.vsphere_virtual_machine.template.scsi_type}"

  network_interface {
    network_id   = "${data.vsphere_network.net.id}"
    adapter_type = "${data.vsphere_virtual_machine.template.network_interface_types[0]}"
  }

  disk {
    label = "${var.hostname_prefix}-${format("%02d", count.index + 1)}.${var.domain}_0"
    size  = "${var.disk_size}"
  }

  clone {
    template_uuid = "${data.vsphere_virtual_machine.template.id}"

    customize {
      linux_options {
        host_name = "${var.hostname_prefix}-${format("%02d", count.index + 1)}"
        domain    = "${var.domain}"
      }

      network_interface {}

    }
  }

  provisioner "remote-exec" {
    script = "${var.bootstrap_script}"

    connection {
      type     = "ssh"
      user     = "${var.ssh_user}"
      password = "${var.ssh_pass}"
    }
  }

}
```

### Debug Output
https://pastebin.com/evaW1hkz

**Comment by swetli** [comment]
I had the same issue. Fixed it by converting the template to VM and setting the VM network to "VM network" since the template was pointing to invalid network. Ofc you have to convert the VM back to template in order to deploy VMs from it.

**Comment by skywalk7** [comment]
This is the duplicate of #316

**Comment by bill-rich** [comment]
Agreed. Fixed in #316.
